File: EPR-GTG-dns-module/bin/dnsdata/org/bsdpanel/dns/Text.java
package org.bsdpanel.dns;

public class Text extends Record {
	protected String contents;
	public Text() {
		contents = "";
	}
    
    public Text(final String text) {
        contents = text;
    }
    
    public Text(final Text text) {
        contents = text.contents;
    }
    
    public String toString() {
        return contents;
    }
    
    public String toBind() {
        return "" + fqdn.toString() + "\tIN TXT \"" + contents + "\"";
    }
    
    public String toDjb() {
        return "";
    }
}

File: bin/dnsdata/src/class/Text.cpp
#include <iostream>
#include <string>

#include "Std.h"
#include "Text.h"

Text::Text() {
    contents = "";
}

Text::Text(const string text) {
    contents = text;
}

#ifdef WITH_BIND
Text::operator<<(ostream& os, const Text& txt) {
    // Text Record    <fqdn>    IN TXT    "text information"
    os << fqdn << "\tIN TXT\t\"" << contents << "\"\n";
}
#endif

#ifdef WITH_DJBDNS
Text::operator<<(ostream& os, const Text& txt) {
    // Text Record   'fqdn:s:ttl:timestamp:lo
    makeSafe(contents);
    os << "'" << fqdn << ":" << contents << ":" << ttl << ":" << timestamp << ":" << lo << "\n";
}
#endif

Text::makeSafe(string& str) {
    // make the string safe for djbdns, convert symbols to octal
}